Dockerfile

```diff
@@ -0,0 +1,39 @@
+FROM docker.io/library/debian:bullseye-slim as base
+ENV SHELL /bin/bash
+
+WORKDIR /opt
+RUN apt-get -y update --fix-missing && \
+    apt-get -y upgrade && \
+    apt-get install -y --no-install-recommends \
+    libssl-dev make cmake graphviz \
+    git pkg-config curl time rhash ca-certificates jq \
+    python3 python3-pip lsof ruby ruby-bundler git-restore-mtime xz-utils zstd unzip gnupg protobuf-compiler \
+    wget net-tools iptables iproute2 iputils-ping ed zlib1g-dev wabt && \
+    curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
+    apt-get install -y nodejs
+
+# foundry
+RUN curl -L https://foundry.paradigm.xyz/ | bash && \
+    ${HOME}/.foundry/bin/foundryup
+# go
+RUN wget https://dl.google.com/go/go1.20.4.linux-amd64.tar.gz && \
+    # -C to move to given directory
+    tar -C /usr/local/ -xzf go1.20.4.linux-amd64.tar.gz
+# pnpm
+RUN npm install pnpm --global
+
+FROM base AS builder
+COPY . /app
+WORKDIR /app
+
+ENV PATH="${PATH}:/usr/local/go/bin"
+ENV PATH="${PATH}:/root/.foundry/bin"
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile
+RUN pnpm run -r build
+
+FROM builder AS store-indexer
```

Comment:
Actually looking at this now you could likely Copy in just the buld artifacts and then run `pnpm i --production` to not install dev deps and have a smaller image. This is a nit though if the image is not too large don't need to optimize now

Reply:
If you want to make it super tiny you would use a smaller image than bullseye slim. You don't need very much to run just a node process usually requires more extra tooling to build an app than to run an app.

As an example I build with bullseye-slim. but then run the application with the much tinier `gcr.io/distroless/nodejs18 as example-server-runner` image here https://github.com/roninjin10/stax/blob/main/Dockerfile#L242

I use distroless specifically because it's debian based